src/middleware/log_request.rs

```diff
@@ -58,6 +58,16 @@ pub fn add_custom_metadata<V: Display>(req: &mut dyn Request, key: &'static str,
     }
 }
 
+#[cfg(test)]
+pub(crate) fn get_log_message(req: &dyn Request, key: &'static str) -> String {
+    for (k, v) in &req.extensions().find::<CustomMetadata>().unwrap().entries {
+        if key == *k {
+            return v.clone();
+        }
+    }
+    String::new()
```

Comment:
Why a `String::new()`? Since this is for tests, if the key is missing I'd expect a panic.

Reply:
Good suggestion.  I've been working on some upstream `conduit` changes and have been thinking in production mode for a lot of that.  I've converted this to a panic.